### DISCUSSION TITLE
Limited Context Extracted from CSV Files by RAG API

### DISCUSSION THREAD

**Discussion by timmanik** [danny-avila/LibreChat#4066] [Troubleshooting] [answered]
I am currently running the `rag_api` service with the following configuration in my `.env` file:

```bash
HF_TOKEN="hf_"
RAG_API_URL="http://host.docker.internal:8000"
EMBEDDINGS_PROVIDER="huggingface"
EMBEDDINGS_MODEL="sentence-transformers/all-MiniLM-L6-v2"
```

I tested the service with both offline and online model providers (Ollama and Google) to answer a question based on an uploaded CSV file containing 51 rows (1 heading + 50 rows listing tech companies).

In both cases, the models were only able to process up to 5 rows of the CSV content. Below are screenshots showing:

**1. Ollama running Llama3**
<img width="1097" alt="image" src="https://github.com/user-attachments/assets/068a00b7-1e61-4ccf-91b8-32b30aa95c30">

**2. Google running Gemini-pro**
<img width="1122" alt="image" src="https://github.com/user-attachments/assets/867b9f51-8db1-4123-a854-b7a1a8b707d0">

Given that these models have a larger context window, I believe the issue is not related to the model's context limit but rather how much of the uploaded file is being included in the request context.

Has anyone experienced a similar issue or can point me to a solution?

**Comment by fuegovic** [comment, marked as answer]
you can add this to your .env :
```
RAG_USE_FULL_CONTEXT=true
```

**Reply by timmanik** [reply]
Thanks that did the trick!

**Reply by beb3k** [reply]
> you can add this to your .env :
> 
> ```
> RAG_USE_FULL_CONTEXT=true
> ```

Could you guys please write this environment variable somewhere on the docs or the rag_api repo readme? Pretty sure others will encounter this issue later on

**Reply by timmanik** [reply]
Thanks for the suggestion, @beb3k. I added a few PRs for it https://github.com/danny-avila/LibreChat/pull/4494
